routing/route/route.go

```diff
@@ -169,6 +177,16 @@ type Route struct {
 	Hops []*Hop
 }
 
+// AddMPP extends existing tlv records destined for the final hop
+// with an option_mpp record use to construct base AMP payments.
+func (r *Route) AddMPP(total lnwire.MilliSatoshi, addr [32]byte) {
```

Comment:
I think it makes more sense to set this at `Route` creation time if possible, to avoid mutation.